Jenkinsfile

```diff
@@ -57,10 +57,29 @@ pipeline {
             }
         }
 
-       stage('Integration Tests') {
+        stage('Publish if Master') {
+            when {
+                expression {
+                    return env.BRANCH_NAME == "master"
+                }
+            }
+            environment {
+                DOCKER = credentials('rook-docker-hub')
+                AWS = credentials('rook-jenkins-aws')
+                GIT = credentials('rook-github')
+            }
+            steps {
+                sh 'docker login -u="${DOCKER_USR}" -p="${DOCKER_PSW}"'
+                sh 'build/run make -j\$(nproc) -C build/release build BRANCH_NAME=${BRANCH_NAME} GIT_API_TOKEN=${GIT_PSW}'
+                sh 'build/run make -j\$(nproc) -C build/release publish BRANCH_NAME=${BRANCH_NAME} AWS_ACCESS_KEY_ID=${AWS_USR} AWS_SECRET_ACCESS_KEY=${AWS_PSW} GIT_API_TOKEN=${GIT_PSW}'
+                sh 'build/run make -j\$(nproc) -C build/release promote BRANCH_NAME=master CHANNEL=master AWS_ACCESS_KEY_ID=${AWS_USR} AWS_SECRET_ACCESS_KEY=${AWS_PSW}'
+            }
+        }
+
+        stage('Integration Tests') {
             when {
                 expression {
-                    return env.shouldBuild != "false" && env.shouldTest != "false"
+                    return env.shouldBuild != "false" && env.shouldTest != "false" && !env.skipIntegrationTests
```

Comment:
the other checks are against string values (e.g., `"false"`), but this new one uses a boolean operator.  is there a reason for that? not an expert on Jenkinsfile syntax.

Reply:
Actually I'm still trying to get this condition to work. I've tried several variations including string comparisons or the boolean and they are always evaluating to false and skipping the stage. I'm a novice at the jenkins syntax as well!